presto-cli/src/main/java/com/facebook/presto/cli/QueryRunner.java

```diff
@@ -60,6 +60,9 @@ public QueryRunner(
                 kerberosConfig,
                 Optional.<JettyIoPool>empty(),
                 ImmutableList.<HttpRequestFilter>of());
+        if (session.getPassword() != null) {
+            ((JettyHttpClient) httpClient).enableLdapAuthentication(session.getServer(), session.getUser(), session.getPassword());
```

Comment:
I don't like this trick with `enableLdapAuthentication`.  I would rather implement `LDAP` authenticaion as a `HttpRequestFilter` and add that filter to `ImmutableList.<HttpRequestFilter>of())`. Only you need is to just add additional headers with authentication to the request you are about to send. So, i see no reason why should it be implemented in such strange way.